### ISSUE TITLE
Sidemenu: Implement correct hover/focus/current state

### ISSUE THREAD

**Issue by ashharrison90** [grafana/grafana#39021]
**Why is this component needed**:

**Situation:** 
- Right now hover and selected states are visually similar – they both employ gradient borders. We aim to visually distinguish these two states as much as possible.
- Grafana currently lacks a clear indicator of which navbar item is selected, and where in the navigation our users are. This makes wayfinding problematic.
- Focus state is clunky. The left border of the blue focus ring spills over the screen. It also lacks prominence. Tabbing is unpredictable, which is why focus states should be more visually reinforced.

**Goals:** 

- Design a hover state that is visually different than the selected state.
- Design a more prominent focused state.
- Make it clear which section of the navbar is selected so that users know where they are in the navigation by designing a selected state.
- Solutions will be consistent with our design system and patterns that Grafana is currently employing, and it will be in line with best practices of UI design.

**Solutions:**

- All states have been checked for accessibility.
- All states employ a double indicator of state change: e.g. overlay + icon color; icon color + border.
- Hover state uses a dark overlay, and the default icon color changes from secondary to primary. It has low visual emphasis as the cursors assists users in tracking where they are.
- Focus state adds a focus ring on top of the hover state. Focus ring is inside the element, it does not spill outside of the navbar. It has medium visual emphasis as tabbing is unpredictable and there is no cursor to assist orientation.
- Selected state utilises our primary icon color and a gradient border on the left side of the element container. This
 is consistent with how we indicate selected tabs in our UI. It has medium emphasis that allows users to quickly understand where they are, but does not take away from the main content.

<img width="426" alt="Screenshot 2021-09-08 at 14 50 26" src="https://user-images.githubusercontent.com/74137331/132512327-f46d4c56-fb88-43ae-b2f1-b8c59041a166.png">
___

This will be used in the navigation sidebar for both the light and the dark theme.
For https://github.com/grafana/grafana/issues/37961

as per the designs from @VesnaDean from [thisFigma file](https://www.figma.com/file/m0q3sBVILBNGZO1QolsMn6/Navbar-explorations?node-id=0%3A1)
